src/Test/Integration/Block/SSR/RecordListTest.php

```diff
@@ -43,6 +50,27 @@ public function test_will_redirect_to_product_page_on_articleNumberSearch()
         $block->toHtml();
     }
 
+    public function test_will_detect_relative_url()
+    {
+        $isAbsoluteUrlMethod = $this->invokeMethod($this->recordList, 'isAbsoluteUrl', ['/relative-url']);
```

Comment:
These are a unit tests. You manually create a tested object detached from the environment, call the method with some imagined argument and assert the result or the execution.  Logic is correct but this should be moved to Unit Tests.
On the other hands because of its access level, these functions should not be unit-tested directly.

The idea of integration tests is to run the test in only partially mocked environment. The majority of it should be real.

So here we can call the object manager to create block that interests us, call  the main method which is _toHtml and execute the assertions against the real data. 

In your tests you manually create the block, and using Reflection just call its private method with the artificially passed argument.